synapse/storage/databases/main/room.py

```diff
@@ -912,7 +912,7 @@ def _get_media_mxcs_in_room_txn(
                 event_json = db_to_json(content_json)
                 content = event_json["content"]
                 content_url = content.get("url")
-                thumbnail_url = content.get("info", {}).get("thumbnail_url")
+                thumbnail_url = (content.get("info") or {}).get("thumbnail_url")
```

Comment:
This only fixes the problem if `info` is falsey (i.e. a JSON `null`, `false`, `0` or `''`). If `info` is any other value that isn't an object (`true`, number, string, array) then the problem still endures.

Ideally, we would be rejecting events with malformed event content like this in the first place.

Reply:
I suggest doing something like

```python
info = content.get("info")
if isinstance(info, dict):
    thumbnail_url = info.get("thumbnail_url")
else:
    thumbnail_url = None
```